### PR TITLE
Auto bf16

### PR DESCRIPTION
**Pull request by xu98bin**
### PR types
New features

### PR changes
APIs

### Describe
添加

### CODE REVIEW

**Review by zhaoyinglia — COMMENTED** [review]
add unittest file.

File: python/paddle/distributed/passes/__init__.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 from .auto_parallel_sharding import *  # noqa: F403
 from .auto_parallel_amp import *  # noqa: F403
 from .auto_parallel_fp16 import *  # noqa: F403
+from .auto_parallel_bf16 import BF16Pass
```
Comment (zhaoyinglia) [inline python/paddle/distributed/passes/__init__.py]: add `# noqa: F403`

File: python/paddle/distributed/auto_parallel/parallelizer_v2.py [hunk 2]
```diff
@@ -20,7 +20,6 @@
 from paddle.fluid.backward import append_backward
 from paddle.fluid.framework import unique_name
 from paddle.distributed.passes import new_pass
-
```
Comment (zhaoyinglia) [inline python/paddle/distributed/auto_parallel/parallelizer_v2.py]: Do not remove this blank line.

File: python/paddle/distributed/auto_parallel/constants.py [hunk 3]
```diff
@@ -76,6 +76,14 @@ def set_field_default_config(category, field, default_value):
 set_field_default_config(AMP, "use_fp16_guard", True)
 set_field_default_config(AMP, "use_optimizer_fp16", False)
 
+set_field_default_config(AMP, "enable_bf16", False)
+set_field_default_config(AMP, "custom_bf16_list", [])
+set_field_default_config(AMP, "custom_fp32_list", [])
+set_field_default_config(AMP, "custom_fp32_varnames", [])
+set_field_default_config(AMP, "use_pure_bf16", False)
+set_field_default_config(AMP, "use_bf16_guard", False)
+set_field_default_config(AMP, "use_optimizer_bf16", False)
```
Comment (zhaoyinglia) [inline python/paddle/distributed/auto_parallel/constants.py]: remove this config.

File: python/paddle/distributed/passes/__init__.py (hunk 1 above)
Comment (zhaoyinglia) [inline python/paddle/distributed/passes/__init__.py]: `from .auto_parallel_bf16 import *`

File: python/paddle/distributed/passes/auto_parallel_bf16.py [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+        ops = self._block.ops
+        dist_op_context = dist_context.dist_op_context
+        training = False
+        for op in ops:
+            if int(op.attr("op_role")) == 257:
+                training = True
+            if op.type == "create_py_reader" or op.type == "read":
+                continue
+            if int(op.attr("op_role")) == int(OpRole.Forward):
+                if self.fp32_varnames is not None and _is_in_fp32_varnames(
```
Comment (zhaoyinglia) [inline python/paddle/distributed/passes/auto_parallel_bf16.py]: This part writes into a function like amp_pass.

File: python/paddle/distributed/auto_parallel/parallelizer_v2.py (hunk 2 above)
Comment (xu98bin) [reply python/paddle/distributed/auto_parallel/parallelizer_v2.py]: I have restoreed it.

File: python/paddle/distributed/passes/__init__.py (hunk 1 above)
Comment (xu98bin) [reply python/paddle/distributed/passes/__init__.py]: ok

File: python/paddle/distributed/auto_parallel/constants.py (hunk 3 above)
Comment (xu98bin) [reply python/paddle/distributed/auto_parallel/constants.py]: have removed this config

File: python/paddle/distributed/passes/__init__.py (hunk 1 above)
Comment (xu98bin) [reply python/paddle/distributed/passes/__init__.py]: If I use "from .auto_parallel_bf16 import *",I can not pass pre-commit.

**Pull request closed without merging**